### ISSUE TITLE
Feature Request: add inline comment for compilation errors by Problem APIs.

### ISSUE THREAD

**Issue by ganadist** [gradle/actions#30]
It seems Gradle is working about [Problems API](https://github.com/gradle/gradle/issues/27596).
And these APIs contains problem events where errors are from.

So, if these APIs can integrated with Github Actions, developers can easily understand where compilation issues are coming from.

**Comment by hfhbd** [comment]
Do you still do not plan this feature in the near feature after integration the Java compiler with the problems api in Gradle 8.9? Or would you accept a contribution?

**Comment by bigdaz** [comment]
@hfhbd I won't have time to work on this in the coming months. A contribution would be welcome.

**Comment by hfhbd** [comment]
> Is there a way to add listener on [init-scripts](https://github.com/gradle/actions/tree/main/sources/src/resources/init-scripts) of setup-gradle action?

Indeed. I did try to implement the support today but the lack of support to add a progress listener does not work, because the `ProblemEvent` is a subtype of the `ProgressEvent` that needs a `ProgressListener` but part of the tooling api, not of Gradle itself.
I will create an issue in gradle/gradle.

**Comment by hfhbd** [comment]
After looking at the html file (and the DefaultProblemsReportCreator), Gradle already emits a JSON representation of the problems inside the html report, that is displayed by the HTML report with JS. But this representation is inside the html file, a dedicated json file would be preferred.

So I created https://github.com/gradle/gradle/issues/32586 to export the json file.
